lib/models/workdir-cache.js

```diff
@@ -66,8 +68,12 @@ export default class WorkdirCache {
           }
 
           if (!stat.isDirectory()) {
-            // A file called ".git". Probably not a good idea, but still not a git working directory.
-            return walk();
+            const contents = await readFile(dotGit, 'utf8');
+            if (contents.startsWith('gitdir: ')) {
+              return resolve(currentDir);
+            } else {
+              return walk();
+            }
```

Comment:
:+1:

Although, I am wondering if it's still worth it to keep this as native filesystem operations as opposed to shelling out to `git rev-parse --show-toplevel`, since it does more than test for `.git` directories.

For one thing, we'll be slightly inconsistent with the command-line in that we won't handle [`GIT_CEILING_DIRECTORIES`](https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables#_repository_locations), for example.